shop/src/components/payment/crypto/Crypto.js

```diff
@@ -17,18 +17,22 @@ import TokenList from './_TokenList'
 const waitForAllowance = ({ wallet, marketplace, amount, token }) => {
   return new Promise((resolve, reject) => {
     wallet.signer.getAddress().then((address) => {
-      function checkAllowance() {
-        token.contract
-          .allowance(address, marketplace.address)
-          .then((allowance) => {
-            const allowanceEth = ethers.utils.formatUnits(allowance, 'ether')
-            if (allowanceEth >= amount) {
-              resolve()
-            } else {
-              setTimeout(checkAllowance, 5000)
-            }
-          })
-          .catch(reject)
+      async function checkAllowance() {
+        try {
+          const decimals = await token.contract.decimals()
+          const allowance = await token.contract.allowance(
+            address,
+            marketplace.address
+          )
+          const allowanceEth = ethers.utils.formatUnits(allowance, decimals)
```

Comment:
Doesn't ethers.utils.formatUnits return a string?
Should it get converted to a number before doing the comparison below?

Reply:
Yeah, it does return a finite number wrapped as string. But the comparisons works because it's JS 😅 

I'm not sure if parsing it as number will break anything else, will make a note of it and change it in a different branch.